app-modules/service-management/src/Observers/ServiceRequestAssignmentObserver.php

```diff
@@ -56,6 +56,8 @@ class ServiceRequestAssignmentObserver
 
     public function creating(ServiceRequestAssignment $serviceRequestAssignment): void
     {
+        $serviceRequestAssignment->loadMissing('serviceRequest');
+
         throw_if(! in_array($serviceRequestAssignment->user_id, $serviceRequestAssignment->serviceRequest->priority->type?->managers
```

Comment:
```suggestion
        throw_if(! in_array($serviceRequestAssignment->user_id, $serviceRequestAssignment->serviceRequest?->priority?->type?->managers
```

For safety, it's probably worth also doing this, in case the service request is truly missing and the load doesn't fix the issue? Also on line 64, adding a `?? []` after the `toArray()` so that the `in_array()` doesn't fix the issue.

However, I guess that `AttemptedToAssignNonManagerToServiceRequest` would actually get thrown in that case, which isn't caught anywhere. It would be really useful to inspect that `$serviceRequestAssignment` object, can it be converted into an array and attached to the exception message?